audio/audio.py

```diff
@@ -97,8 +99,12 @@ def sr(self, sr): self._sr = sr
     def ipy_audio(self): return Audio(data=self.sig, rate=self.sr)
 
     @property
-    def duration(self): return len(self.sig)/self.sr
-
+    def duration(self): 
+        if(self._sig is not None and self._sr is not None): return len(self.sig)/self.sr
```

Comment:
You can just call len(self.sig)/self.sr the getter methods will solve the rest

Reply:
self.sig will read in the entire signal, we just need the duration, but you're right about self.sr so I've removed the self._sr is not None check, and I also made a change to save .sr in the .sr getter (it was just checking and returning)